pkg/policies/loader.go

```diff
@@ -42,35 +50,58 @@ func (e *EmbeddedLoader) Load(_ context.Context, attachment *v1.PolicyAttachment
 	return attachment.GetEmbedded(), nil
 }
 
-// BlobLoader loader loads policies from filesystem and HTTPS references using Cosign's blob package
-type BlobLoader struct{}
+// FileLoader loader loads policies from filesystem and HTTPS references using Cosign's blob package
+type FileLoader struct{}
 
-func (l *BlobLoader) Load(_ context.Context, attachment *v1.PolicyAttachment) (*v1.Policy, error) {
+func (l *FileLoader) Load(_ context.Context, attachment *v1.PolicyAttachment) (*v1.Policy, error) {
 	var (
 		rawData []byte
 		err     error
 	)
 
-	reference := attachment.GetRef()
+	ref := attachment.GetRef()
+	filePath, err := ensureScheme(ref, fileScheme)
+	if err != nil {
+		return nil, err
+	}
 
-	// Support file:// references
-	parts := strings.SplitAfterN(reference, "://", 2)
-	if len(parts) == 2 && parts[0] == "file://" {
-		rawData, err = os.ReadFile(filepath.Clean(parts[1]))
-		if err != nil {
-			return nil, fmt.Errorf("loading policy spec: %w", err)
-		}
+	rawData, err = os.ReadFile(filepath.Clean(filePath))
+	if err != nil {
+		return nil, fmt.Errorf("loading policy spec: %w", err)
 	}
 
-	// this method understands env, http and https schemes, and defaults to file system (without scheme).
-	if rawData == nil {
-		rawData, err = blob.LoadFileOrURL(reference)
-		if err != nil {
-			return nil, fmt.Errorf("loading policy spec: %w", err)
-		}
+	return unmarshalPolicy(rawData, filepath.Ext(ref))
+}
+
+// HTTPSLoader loader loads policies from HTTP or HTTPS references
+type HTTPSLoader struct{}
+
+func (l *HTTPSLoader) Load(_ context.Context, attachment *v1.PolicyAttachment) (*v1.Policy, error) {
+	ref := attachment.GetRef()
+
+	var httpRef string
```

Comment:
these two variables could be replaced if you run the following stanza in two lines

```
	if httpRef, err = ensureScheme(ref, httpScheme, httpsScheme); err != nil {
```

```
httpRef, err := ensureScheme(ref, httpScheme, httpsScheme)
if err != nil {
```